### PR TITLE
Added firebase messaging to iOS

### PR DESCRIPTION
**Pull request by cindy-x-liang**
Integrated firebase cloud messaging for iOS so that I could test backend notifs with iOS

### CODE REVIEW

File: TCAT/Base/AppDelegate.swift [hunk 1]
```diff
@@ -90,13 +91,49 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        UNUserNotificationCenter.current().delegate = self
+
+        let authOptions: UNAuthorizationOptions = [.alert, .badge, .sound]
+        UNUserNotificationCenter.current().requestAuthorization(
+          options: authOptions,
+          completionHandler: { _, _ in }
+        )
+
+        application.registerForRemoteNotifications()
+
```
Comment (rs929) [inline TCAT/Base/AppDelegate.swift]: Delete whitespace here

File: TCAT/Base/AppDelegate.swift [hunk 2]
```diff
@@ -90,13 +91,49 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        UNUserNotificationCenter.current().requestAuthorization(
+          options: authOptions,
+          completionHandler: { _, _ in }
+        )
+
+        application.registerForRemoteNotifications()
+        
+        
+        Messaging.messaging().delegate = self
+
```
Comment (rs929) [inline TCAT/Base/AppDelegate.swift:105]: Delete all whitespace, but leave one before the return

File: TCAT/Base/AppDelegate.swift [hunk 3]
```diff
@@ -90,13 +91,49 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        Messaging.messaging().token { token, error in
+          if let error = error {
+            print("Error fetching FCM registration token: \(error)")
+          } else if let token = token {
+            print("FCM registration token: \(token)")
+            
+          }
+        }
+           
+       }
```
Comment (rs929) [inline TCAT/Base/AppDelegate.swift]: Looks like the indentation is wrong here for the brackets, make sure to format correctly, you can do this by selecting the code, and then hitting ctrl (not command)+ i

File: TCAT/Base/AppDelegate.swift [hunk 4]
```diff
@@ -206,3 +243,31 @@ extension UIWindow {
...
 
 }
+
+extension AppDelegate {
+
+//MessagingDelegate
+//    func messaging(_ messaging: Messaging, didReceiveRegistrationToken fcmToken: String?) {
+//        self.firebaseToken = fcmToken!
+//        print("Firebase token: \(fcmToken)")
+//    }
```
Comment (rs929) [inline TCAT/Base/AppDelegate.swift]: If this code is not needed, plz remove

File: TCAT/Base/AppDelegate.swift [hunk 5]
```diff
@@ -206,3 +243,31 @@ extension UIWindow {
...
+    
+    func messaging(_ messaging: Messaging, didReceiveRegistrationToken fcmToken: String?) {
```
Comment (rs929) [inline TCAT/Base/AppDelegate.swift:240]: Looks like indentation for this function is off as well lol, do the same as above ^

**Review by rs929 — APPROVED** [review]
Nice PR! Just a few style issues with whitespace and indentation, plz fix before merging.

Great work Cindy!

**Review by JaysonHahn — APPROVED** [review]
LGTM

**Pull request merged**